packages/cozy-viewer/src/Panel/helpers.js

```diff
@@ -0,0 +1,43 @@
+import { splitFilename } from 'cozy-client/dist/models/file'
+
+/**
+ * Returns file extension or class
+ * @param {import("cozy-client/types").IOCozyFile} file - io.cozy.file
+ * @returns {string}
+ */
+export const makeFormat = file => {
+  const { extension } = splitFilename(file)
+  return (extension.replace('.', '') || file.class).toUpperCase()
+}
+
+/**
+ * Returns a formatted date
+ * @param {string} lang - language in ISO 639-1 format
+ * @returns {string}
+ */
+export const makeDate = lang =>
+  lang === 'fr' ? 'DD MMM YYYY à HH:mm' : 'MMM DD YYYY at HH:mm'
+
+/**
+ * Returns a formatted size
+ * @param {number} bytes - file bytes
+ * @returns {string}
+ */
+export const makeSize = bytes => {
```

Comment:
je n'ai pas trouvé côté cozy-client, je suis parti sur du custom du coup. Après vérification dans Drive, on utilise une lib `filesize` https://www.npmjs.com/package/filesize pour ça. Comme ça tient en 4 lignes, pas sûr que la lib se justifie 🤔

Reply:
My point was to make it shared via cozy-client rather than adding a custom one. You don't have all your functions ready-made, but some of them overlap with the model/intance functions ([example of code](https://github.com/cozy/cozy-client/blob/15084c7361c5eda07db93d50dc1cd1a124dabadc/packages/cozy-client/src/models/instance.js#L124)). We could divide yours into several helpers so as not to repeat the same type of work. I agree for 4 line the lib is not justified in cozy-drive. 

What's more, you'll save time when you want to migrate the FolderPicker inside cozy-ui because it needs this function. Now, its done with the lib `filesize` (code)[https://github.com/cozy/cozy-drive/blob/e442c4b6545a364d1c832aef8973e5b4b6679bc8/src/components/FolderPicker/FolderPickerListItem.tsx#L40] 🙂

Reply:
Ok je verrai à ce moment là. Si on doit extraire le FolderPicker et qu'il y a besoin de mutualiser cette méthode, je le ferai à ce moment là. Le chantier du nouveau viewer est conséquent, je pense que c'est bien de noter ces points de refacto afin d'améliorer la base de code et de mutualiser autant que possible, mais on navigue plutôt à vue sur cette feature qu'on souhaite sortir au fil de l'eau et tester tôt. A mon avis ce genre de retour sont plutôt à traiter dans un deuxième temps, profiter d'une phase de recette par exemple pour faire la refacto en parallèle.